### PR TITLE
Notes from lesson 24.03.2023 - streams and inheritance

### PR DESCRIPTION
**Pull request by PaniOdProgramowania**

### CODE REVIEW

File: src/zoo/Runner.java [hunk 1]
```diff
@@ -5,28 +5,19 @@
...
-            if (animal instanceof Dog) {
-                String dogsVoice = ((Dog) animal).giveVoice();
-                System.out.println(dogsVoice);
-            }
-            if (animal instanceof Tiger) {
-                String tigerVoice = ((Tiger) animal).giveVoice();
-            }
+
+        for (Animal animal : zoo) {
+            System.out.println(animal.getClass().getSimpleName() + " gives voice: " + animal.giveVoice());
```
Comment (PaniOdProgramowania) [inline src/zoo/Runner.java]: 🎉

File: src/zoo/Runner.java [hunk 2]
```diff
@@ -14,7 +14,13 @@ public static void main(String[] args) {
         zoo.add(dog);
         zoo.add(tiger);
 
-
+//        first version:
+        for(int i = 0; i < zoo.size(); i++){
+            Animal animal = zoo.get(i);
+            System.out.println(animal.getClass().getSimpleName() +" gives voice: " + animal.giveVoice());
```
Comment (PaniOdProgramowania) [inline src/zoo/Runner.java]: why `getClass.getSimpleName` ;)? we have `toString` method for such use cases. We rarely should need the class's name. It's checking the type whereas in object oriented programming, the type shouldn't be checked very often. We will learn polymorphism and we'll learn that a `.` is more important than the type :)

File: src/zoo/Animal.java [hunk 3]
```diff
@@ -5,6 +5,10 @@ public class Animal {
     private int age;
     private boolean isWild;
 
+    public boolean getIsWild() {
```
Comment (PaniOdProgramowania) [inline src/zoo/Animal.java]: in such getters, we tend to just name them `isWild` (omitting `get` prefix) :) this way the code reads `animal.isWild` as in `animal: is it wild? yes or no`. It's more English then :)

File: src/zoo/Animal.java [hunk 4]
```diff
@@ -16,4 +20,11 @@ public Animal(String name, int age, boolean isWild) {
     public String giveVoice(){
         return "usual Animal voice ";
     }
+    public String doVildThings(boolean isWild){
```
Comment (PaniOdProgramowania) [inline src/zoo/Animal.java]: I think the name should be `wild` ;) not `vild` :)

File: src/zoo/Animal.java [hunk 5]
```diff
@@ -16,4 +20,11 @@ public Animal(String name, int age, boolean isWild) {
...
+        if(isWild){
+            return "Doing vild things";
+        }else {
+            return "nice " + getClass().getSimpleName();
```
Comment (PaniOdProgramowania) [inline src/zoo/Animal.java]: please don't use `getClass().getSimpleName()` ;) use `toString` in such cases 
https://stackoverflow.com/questions/36828050/is-it-bad-practice-to-use-object-getclass-in-java

**Review by PaniOdProgramowania — COMMENTED** [review]
great work! Just some things for you to consider before we move on:)

File: src/zoo/Runner.java (hunk 2 above)
Comment (marcingas) [reply src/zoo/Runner.java]: ok, done:)

File: src/zoo/Animal.java (hunk 3 above)
Comment (marcingas) [reply src/zoo/Animal.java]: ok, my mistake, done:)

File: src/zoo/Animal.java (hunk 4 above)
Comment (marcingas) [reply src/zoo/Animal.java]: true:), done

File: src/zoo/Animal.java (hunk 5 above)
Comment (marcingas) [reply src/zoo/Animal.java]: yes, never ever :)

File: src/zoo/Animal.java [hunk 6]
```diff
@@ -3,28 +3,37 @@
 public class Animal {
     private String name;
     private int age;
-    private boolean isWild;
+    private boolean Wild;
```
Comment (PaniOdProgramowania) [inline src/zoo/Animal.java]: please keep the field's name lower case :) `W` -> `w`

**Review by marcingas — APPROVED** [review]
ok, done

**Pull request merged**